src/main.js

```diff
@@ -18,154 +19,198 @@ import {
   getChangesetEntriesMap,
   getChangesetFileContent,
   isSkipEntry,
+  isGitHubAppInstalledOrNotSuspended,
   formatPostComment,
 } from "./utils/index.js";
 
-async function run() {
-  // Initialize Octokit client with the GitHub token
-  const octokit = authServices.getOctokitClient();
+import { GitHubAppSuspendedOrNotInstalledWarning } from "./warnings/index.js";
+
+// ****************************************************************************
+// I) MAIN
+// ****************************************************************************
 
-  const changesetFilePath = (prNumber) => `${CHANGESET_PATH}/${prNumber}.yml`;
-  let baseOwner,
-    baseRepo,
-    baseBranch,
-    headOwner,
-    headRepo,
-    headBranch,
-    prNumber,
-    prDescription,
-    prLink;
+const run = async () => {
+  const octokit = authServices.getOctokitClient();
+  let prData = extractPullRequestData();
 
   try {
-    // Step 0 - Extract information from the payload
-    ({
-      baseOwner,
-      baseRepo,
-      baseBranch,
-      headOwner,
-      headRepo,
-      headBranch,
-      prNumber,
-      prDescription,
-      prLink,
-    } = extractPullRequestData());
-
-    // Step 1 - Parse changelog entries and validate
-    const changelogEntries = extractChangelogEntries(
-      prDescription,
-      processChangelogLine
-    );
-    const changesetEntriesMap = getChangesetEntriesMap(
-      changelogEntries,
-      prNumber,
-      prLink
-    );
+    if (!(await isGitHubAppInstalledOrNotSuspended(prData))) {
+      await handleGitHubAppInstalledOrNotSuspended(octokit, prData);
+      return;
+    }
+    await processChangelogEntries(octokit, prData);
+  } catch (error) {
+    await handleErrorChangelogEntries(error, octokit, prData);
+    throw new Error("Changeset creation workflow failed.");
+  }
+};
+
+run();
+
+// ****************************************************************************
+// II) HELPERS
+// ****************************************************************************
+
+
+
+// ----------------------------------------------------------
+// GitHub App Helpers Functions
+// ----------------------------------------------------------
+const handleGitHubAppInstalledOrNotSuspended = async (octokit, prData) => {
```

Comment:
I assume this is temporary and that this flow will only show the warning comment if the cahngeset file isnt present right? after validation

Reply:
Yeah, that is the idea. The validation of the manual changeset file created is the next thing I have to work on.

Reply:
@ashwin-pc,

Refactoring in main.js is finished. Now it is much more cleaner to read. Let me know how it goes.